linter.py

```diff
@@ -12,5 +16,57 @@ class Phpmd(Linter):
     tempfile_suffix = 'php'
     defaults = {
         'selector': 'source.php, text.html.basic',
-        '@rulesets:,': 'cleancode,codesize,controversial,design,naming,unusedcode'
     }
+
+    default_ruleset = 'cleancode,codesize,controversial,design,naming,unusedcode'
+
+    def cmd(self):
+
+        settings = Linter.get_view_settings(self)
+        rulesets = settings.get('rulesets')
+
+        if rulesets.endswith(".xml"):
+            rulesets_file = self.get_conf_file(rulesets)
+            if rulesets_file:
+                return ('phpmd', '${temp_file}', 'text', rulesets_file)
+
+            logger.error('file not found - %s' % rulesets)
+            rulesets = self.default_ruleset
+
+        return ('phpmd', '${temp_file}', 'text', rulesets)
+
+    def get_conf_file(self, filename):
+
+        if os.path.basename(filename) is not filename:
+
+            if not os.path.isabs(filename):
+                settings = self.get_view_settings()
+                working_dir = self.get_working_dir(settings)
+
+                # Resolve path relative to working dir
+                filename = os.path.abspath(os.path.join(working_dir, filename))
+
+            # Absolute file path
+            return os.path.isfile(filename) and filename
+
+        if not self.filename:
+            # The file is not yet saved
+            return False
+
+        path = os.path.dirname(self.filename)
+
+        # Search for the file in directory tree
+        while True:
+            newpath = os.path.dirname(path)
+
+            if path == newpath:
+                return False
+
+            file = os.path.join(path, filename)
```

Comment:
There is a logic/flow error here. E.g. filename is `D:\foo.php` and `D:\phpmd.xml` actually exists.

Reply:
Fixed.